src/setup/kmd/config.rs

```diff
@@ -0,0 +1,86 @@
+//! Utilities for kmd configuration.
+
+use std::{
+    net::SocketAddr,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+use anyhow::{anyhow, Result};
+use tokio::time::{sleep, timeout, Duration};
+
+use crate::setup::{
+    constants::LOAD_FILE_TIMEOUT_SECS,
+    kmd::constants::{KMD_DIR, REST_ADDR_FILE, TOKEN_FILE},
+};
+
+/// Startup configuration for the kmd daemon.
+#[derive(Debug, Clone, Default)]
+pub struct KmdConfig {
+    /// The kmd's directory of the running node.
+    pub path: PathBuf,
+    /// The REST API socket address of the kmd instance.
+    pub rest_api_addr: Option<SocketAddr>,
+    /// Security token needed for the REST API authentication.
+    pub token: String,
+}
+
+impl KmdConfig {
+    /// Creates a new [KmdConfig].
+    pub async fn new(node_path: &Path) -> anyhow::Result<Self> {
+        let mut token = String::new();
+
+        let path = node_path.join(KMD_DIR);
+        if !path.exists() {
+            return Err(anyhow!("couldn't find the {:?} directory", path));
+        }
+
+        timeout(LOAD_FILE_TIMEOUT_SECS, async {
```

Comment:
Just to think about - elegant solution for waiting on files to be created (or deleted ;)) can be fs-notify mechanisms like https://docs.rs/notify/latest/notify/